File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.py
def solution(num):
    step = 0
    
    if num == 1:
        return step
    
    while step < 500:
        if num == 1:
            return step
        
        if num % 2 == 0:
            num = num // 2
            step += 1
            continue
            
        if num % 2 == 1:
            num = num * 3 + 1
            step += 1
            continue
            
    return -1
        

        

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.js
function solution(num) {
    let step = 0;
    while(num !== 1 && step < 500){
        if(num % 2 === 0){
            num = num / 2;
        }else{
            num = num * 3 + 1;
        }
        step++;
    }
    return num !== 1 ? -1 : step;
}